[global] Changed raw "header('Location:') to Friendica's System::externalRedirect()

Quix0r opened the pull request:
Changes:
- rewrote `header('Location: foo');` to `System::externalRedirect()`
- changed double-quotes to single
- no executable permission on these files

Fixed an issue reported in https://github.com/friendica/friendica/issues/11632#issuecomment-1237755268

Quix0r committed:
Changes:
- rewrote `header('Location: foo');` to `System::externalRedirect()`
- no executable permission on these files

MrPetovan commented on saml/saml.php:
```diff
@@ -119,10 +120,7 @@ function saml_sso_initiate(App $a, array &$b)
 	$ssoBuiltUrl = $auth->login(null, [], false, false, true);
 	$_SESSION['AuthNRequestID'] = $auth->getLastRequestID();
-	header('Pragma: no-cache');
-	header('Cache-Control: no-cache, must-revalidate');
```
Please keep the cache control headers, I’m not sure what would be the side-effects of removing them.

Quix0r replied:
Or cannot we move them to the method `System::externalRedirect()`?

annando replied:
Moving the headers to the function could have got other side effects.

Quix0r replied:
Okay, I would like to have this easier for addon developers, like they don't need to care so much by themselves about it. I will add another parameter to the method.

Quix0r replied:
`require_once(__DIR__ . '/vendor/autoload.php');` <-- Can this line be removed?

MrPetovan commented on saml/saml.php:
```diff
@@ -182,13 +180,9 @@ function saml_slo_initiate(App $a, array &$b)
 	$sloBuiltUrl = $auth->logout();
 	$_SESSION['LogoutRequestID'] = $auth->getLastRequestID();
-	header('Pragma: no-cache');
-	header('Cache-Control: no-cache, must-revalidate');
```
Please keep the cache control headers.

Quix0r committed:
Changes:
- added 3rd parameter `$proxy_headers` to add (default: not added)
  Pragma/Cache-Control headers
- But maybe there is a better one already in some composer package, that also
  supports more than just these 2 headers?

Quix0r commented:
This change requires https://github.com/friendica/friendica/pull/11900 to be merged or else this won't add those headers.
